src/Core/Utils/BijectiveAssociation.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <initializer_list>
+#include <map>
+
+namespace Ra {
+namespace Core {
+namespace Utils {
+
+/**
+ * \brief Bijective association on a finite set of <key, value> pairs.
+ * \tparam T1 Type of one element of the pair
+ * \tparam T2 Type of the other element of the pair
+ * \note T1 must be different from T2
```

Comment:
Do this limit comes only from operator() ?
There is currently a bijective association in `Engine::Mesh` which would benefit of this new class. But it's string <> string.
```c++
    /// Core::Mesh attrib name to Render::Mesh attrib name
    using TranslationTable = std::map<std::string, std::string>;
    TranslationTable m_translationTableMeshToShader;
    TranslationTable m_translationTableShaderToMesh;
```

Reply:
This limit is not only from operator () but also from 'addAssociation 'and the constructors. 
If the two types are the same, compiler generates an error such as `error: multiple overloads of 'addAssociation' instantiate to the same signature`.
This can be solved by removing the capability to add pair whatever their elements' order (e.g. {T1, T2} or {T2, T1}) and restricting the construction using only pari (or parameters in the order {T1, T2}.
We can also change the operators () to two different operator names, e.g inject : T1 -> T2 and surject : T2 -> T1, instead of overloading.
But if types are different, I found very practical to have the () operator.
Perhaps we can rewrite this class using std::enable_if<std::is_same<T1, T2> .... > to define only the alowed functions. I'll try to do that.

Reply:
done using partial specialisation ( I do not success in using std::enable_if :( ). 
If you want to port your use case, you can  push your commit here or wait another PR specific to your fix.

Reply:
Well the PR is big enough to wait for Engine::Mesh.